Makefile

```diff
@@ -28,111 +28,124 @@
 
 ANDROID_STANDALONE_TOOLCHAIN_PATH ?= /usr/local/toolchain
 
+subbuilddir:=$(shell echo  `uname | sed -e 's|[:/\\ \(\)]|_|g'`/`git branch | grep '\* ' | cut -f2- -d' '| sed -e 's|[:/\\ \(\)]|_|g'`)
+ifeq ($(USE_SINGLE_BUILDDIR),)
+  builddir="build/$(subbuilddir)"
+else
+  builddir="build"
+endif
```

Comment:
This introduces different levels of subdirectories, but the number of `../` is fixed. So for me, it doesn't work when using USE_SINGLE_BUILDDIR=1

Reply:
My fault too I would have to test it...